### ISSUE TITLE
implementation is not matching the cwnp spec

### ISSUE THREAD

**Issue by mwennrich** [metal-stack/firewall-controller#73]
CRD describes spec.egress.port as `numerical or named port` ( https://github.com/metal-stack/firewall-controller/blob/master/config/crd/bases/metal-stack.io_clusterwidenetworkpolicies.yaml#L65-L72 ) while in validator `only int ports are supported` ( https://github.com/metal-stack/firewall-controller/blob/714a5bf22625c2f17e7f30a8db77b653cfb63eaa/api/v1/clusterwidenetworkpolicy_types.go#L128-L133 )

**Comment by GrigoriyMikhalkin** [comment]
@mwennrich If i'm correct, `egress` and `ingress` fields specify IPs(or subnets) outside of cluster, from which traffic is allowed. Does it even makes sense to allow to specify named port? Probably a better solution would be to replace `NetworkPolicyPort` fields with custom port field that would be always `int`.

**Comment by majst01** [comment]
egress is for communication with outside endpoints, ingress for communication with inside endpoints.

**Comment by GrigoriyMikhalkin** [comment]
@majst01 Yep, i got that) What i meant is that, both for `egress` and `ingress` rules we specify addresses outside of cluster, so named ports don't even make sense for that case. So, for cleanliness, it's probably better to introduce our custom structure for ports.

**Comment by GrigoriyMikhalkin** [comment]
Maybe i misunderstood this. Comment to `ports` field says:
> List of ports which should be made accessible on the cluster for this rule

So, does it mean that `ports` field lists ports of services in cluster?

**Comment by majst01** [comment]
So at least description must be adopted

**Comment by Gerrit91** [comment]
@mwennrich Do we still want to do this or can we close?

**Comment by mwennrich** [comment]
it's still wrong:
```
$ k explain cwnp.spec.egress.ports
GROUP:      metal-stack.io
KIND:       ClusterwideNetworkPolicy
VERSION:    v1

(...)

  port  <Object>
    port represents the port on the given protocol. This can either be a
    numerical or named
    port on a pod. If this field is not provided, this matches all port names
    and
    numbers.
    If present, only traffic on the specified protocol AND port will be matched.


```

**Comment by Gerrit91** [comment]
Okay, as docs are inherited because we use `NetworkPolicyPort` from `k8s.io/api/networking/v1`, we cannot simply change the doc string for it.

I'll open a PR.
